src/ast/passes/semantic_analyser.cpp

```diff
@@ -1301,10 +1301,16 @@ void SemanticAnalyser::visit(Call &call)
         } else {
           strlen = integer->value + 1; // Storage for NUL byte.
         }
-      } else if (auto *integer = dynamic_cast<NegativeInteger *>(
-                     call.vargs.at(1).as<NegativeInteger>())) {
+      }
+
+      if (auto *integer = dynamic_cast<NegativeInteger *>(
```

Comment:
Shouldn't this be an `} else if` chain? Otherwise it will inject the cast into the fixed `Integer` parameter above also?